commons/src/main/java/org/infinispan/commons/util/IntSet.java

```diff
@@ -0,0 +1,270 @@
+package org.infinispan.commons.util;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.PrimitiveIterator;
+import java.util.Set;
+import java.util.function.IntConsumer;
+
+import org.infinispan.commons.marshall.MarshallUtil;
+
+/**
+ * Represent a set of integers (e.g. segments) as a {@code BitSet}.
+ *
+ * @author Dan Berindei
+ * @since 9.0
+ */
+public class IntSet implements Set<Integer> {
+   private final BitSet bitSet;
+
+   public static IntSet of(int i1) {
+      IntSet set = new IntSet();
+      set.set(i1);
+      return set;
+   }
+
+   public static IntSet of(int i1, int i2) {
+      IntSet set = new IntSet();
+      set.set(i1);
+      set.set(i2);
+      return set;
+   }
+
+   public static IntSet of(int i1, int i2, int i3) {
+      IntSet set = new IntSet();
+      set.set(i1);
+      set.set(i2);
+      set.set(i3);
+      return set;
+   }
+
+   public static IntSet of(int... elements) {
+      IntSet set = new IntSet();
+      for (int i : elements) {
+         set.set(i);
+      }
+      return set;
+   }
+
+   public IntSet() {
+      bitSet = new BitSet();
+   }
+
+   public IntSet(int initialCapacity) {
```

Comment:
I see what you mean now, I was thinking about capacity as the available range but that's obviously not what it means for the other collections. I'm changing the parameter name to `initialRange`.